ChatClient/src/main/java/ru/online/controllers/MainSceneController.java

```diff
@@ -91,47 +97,36 @@ private void btnSend(ActionEvent actionEvent) {
     }
 
     private void sendMessage() {
-        String message = input.getText().trim();
-        if (message.length() > 0) {
-            MessageDTO dto = new MessageDTO();
-            dto.setMessageType(checkMessageType(message));
-            if (dto.getMessageType().equals(MessageType.PRIVATE_MESSAGE)) {
-                dto.setTo((message.split(" ", 3))[1]);
-                dto.setBody((message.split(" ", 3))[2]);
-            } else {
-                dto.setBody(message);
-            }
-            System.out.println(dto.toString());
-            messageService.sendMessage(dto.convertToJson());
-            input.clear();
+        String msg = input.getText().trim();
+        if (msg.length() == 0) return;
+
+        MessageDTO dto = new MessageDTO();
+        String selected = (String) onlineUsers.getSelectionModel().getSelectedItem();
+        if (selected.equals(ALL)) dto.setMessageType(MessageType.PUBLIC_MESSAGE);
+        else {
+            dto.setMessageType(MessageType.PRIVATE_MESSAGE);
+            dto.setTo(selected);
         }
-    }
 
-    private MessageType checkMessageType(String message) {
-        if (message.startsWith("/w")) {
-            return MessageType.PRIVATE_MESSAGE;
-        }
-        return MessageType.PUBLIC_MESSAGE;
+        dto.setBody(msg);
+        messageService.sendMessage(dto.convertToJson());
+        input.clear();
+
     }
 
-    private void showMessage(String message) {
-        chatArea.appendText(message + System.lineSeparator());
+    private void showMessage(MessageDTO message) {
+        String msg = String.format("[%s] [%s] -> %s\n", message.getMessageType(), message.getFrom(), message.getBody());
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.

Reply:
Thanks, I'll take that into account

Reply:
Happy to help :)